app/frontend/src/features/profile/view/ReferenceList.tsx

```diff
@@ -0,0 +1,50 @@
+import { List, makeStyles } from "@material-ui/core";
+import useUsers from "features/userQueries/useUsers";
+import { ListReferencesRes } from "pb/references_pb";
+
+import ReferenceListItem from "./ReferenceListItem";
+
+const useStyles = makeStyles((theme) => ({
+  referencesList: {
+    "& > *": {
+      paddingBlockEnd: theme.spacing(3),
+    },
+    width: "100%",
+  },
+}));
+
+interface ReferenceListProps {
+  isReceived?: boolean;
+  referencePages: ListReferencesRes.AsObject[];
+  referenceUsers: ReturnType<typeof useUsers>["data"];
+}
+
+export default function ReferenceList({
+  isReceived,
+  referencePages,
+  referenceUsers,
+}: ReferenceListProps) {
+  const classes = useStyles();
+
+  return (
+    <List className={classes.referencesList}>
+      {referencePages
+        .map((page) =>
+          page.referencesList.map((reference) => {
+            const userToShow = referenceUsers?.get(
+              isReceived ? reference.fromUserId : reference.toUserId
+            );
+            return userToShow ? (
+              <ReferenceListItem
+                key={reference.referenceId}
+                isReceived
+                user={userToShow}
+                reference={reference}
+              />
+            ) : null;
```

Comment:
Yeah at the moment it's all users or loading at the list level from the parent. Do you think showing users as soon as possible rather than waiting for all might be better user experience? Might give the perception that it's not taking as long to load 🤔

Reply:
Actually this wouldn't work if I don't use a blocking loading indicator in the parent that uses this, since it is using `useUsers` and there's no way to tell which individual query is still loading so we might end up with the case where the cached user will display correctly, but the error'ed users (let's say user has dodgy internet) will be stuck on an infinite loading state. Think I'll just leave it for now and we can re-visit if this becomes a UX issue when more users test this out.